packages/dashboards/src/lib/services/url-interaction.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from "@angular/core";
+
+@Injectable({
+    providedIn: "root",
+})
+export class UrlInteractionService {
+    public template(url: string, data: any): string{
+        const regex = new RegExp(/(\$\{[a-zA-Z0-9.]*\})/g)
```

Comment:
you could have actually used capturing group for the content not for the whole expression, that would save you the slice (and it would be driven by the regex so if this ever changed you wouldn't need to manipulate the value "manually", just use the value from first captured group

Reply:
Not really sure how to do it, also I am capturing it also with parenthesis so I could replace whole expression.